### PR TITLE
added edge cases tests

### PR DESCRIPTION
**Pull request by xialiu1988**
- [x] Read all of these instructions carefully. Name things exactly as described.
Do all your work in a public repository called data-structures-and-algorithms, with a well-formatted, detailed top-level README.md.
Create a new branch in your repo called ll_insertions.
Your top-level readme should contain a “Table of Contents” navigation to all of your challenges and implementations so far. (Don’t forget to update it!)
Place this implementation in your Data-Structures folder within your repository.
- [x] .append(value) which adds a new node with the given value to the end of the list
- [x] .insertBefore(value, newVal) which add a new node with the given newValue immediately before the first value node
- [x] .insertAfter(value, newVal) which add a new node with the given newValue immediately after the first value node
- [x] Can successfully add a node to the end of the linked list
Can successfully add multiple nodes to the end of a linked list
Can successfully insert a node before a node located i the middle of a linked list
Can successfully insert a node before the first node of a linked list
Can successfully insert after a node in the middle of the linked list
Can successfully insert a node after the last node of the linked list

### CODE REVIEW

File: Data-Structures/src/main/java/Data/Structures/LinkedList/Llist.java [hunk 1]
```diff
@@ -69,50 +69,64 @@ public void append(int val) {
...
             }
-            current = current.next;
-        }
+            while (current != null) {
+                if (current.next.value == value) {
+                    Node newNode = new Node(newVal);
+                    newNode.next = current.next;
+                    current.next = newNode;
+                    return;
+                }
```
Comment (evanslaton) [inline Data-Structures/src/main/java/Data/Structures/LinkedList/Llist.java]: `} else {`

File: Data-Structures/src/main/java/Data/Structures/LinkedList/Llist.java [hunk 2]
```diff
@@ -69,50 +69,64 @@ public void append(int val) {
...
+                else {
+                    if (current.next != null) {
+                        current = current.next;
+                    } else throw new IllegalArgumentException("Value not Exsits!");
+                }
+            }
 
+        }
+        else
```
Comment (evanslaton) [inline Data-Structures/src/main/java/Data/Structures/LinkedList/Llist.java]: `} else {`

File: Data-Structures/src/test/java/Data/Structures/LinkedList/LlistTest.java [hunk 3]
```diff
@@ -176,4 +177,34 @@ public  void insertAfterLastNodeOfLlist(){
...
+        ll.insertBefore(55, 5);
+    }
+
+    @Test(expected = NullPointerException.class )
+    public void testInsertBeforeLlistValueNotExsits() {
+        Llist ll = new Llist();
+        ll.insert(76);
+        ll.insertBefore(55, 5);
+    }
+
```
Comment (evanslaton) [inline Data-Structures/src/test/java/Data/Structures/LinkedList/LlistTest.java]: Extra empty lines
